### ISSUE TITLE
Pod with initcontainer runs into `TopologyAffinityError`

### ISSUE THREAD

**Issue by xrmzju** [kubernetes/kubernetes#110306]
### What happened?

pod failed to pass topologymanager  Admit function
<img width="764" alt="image" src="https://user-images.githubusercontent.com/27884109/171174950-c5ef379a-8bf8-4e88-9535-efe49b650345.png">

### What did you expect to happen?

pod success to start

### How can we reproduce it (as minimally and precisely as possible)?

1. kubelet run with `topologyManagerPolicy: single-numa-node` and `cpuManagerPolicy: static`, and we reserve `3000m` cpu for kubelet and system
2. node info:  2 numa, each numa has 16 cores
3. pod info: 1 initcontainer request 1C and 1 container request 14C 
```yaml
apiVersion: v1
kind: Pod
metadata:
  generateName: nginx-74dd79c47d-
  labels:
    app: nginx
  name: nginx-74dd79c47d-hxv45
  namespace: default
spec:
  containers:
  - command:
    - sh
    - -c
    - 'trap : TERM INT; sleep 100000000000 & wait'
    image: <image1>
    imagePullPolicy: IfNotPresent
    name: container-1
    resources:
      limits:
        cpu: "14"
        memory: 1Gi
      requests:
        cpu: "14"
        memory: 1Gi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
  initContainers:
  - command:
    - sh
    - -c
    - echo hello
    image: <image2>
    imagePullPolicy: IfNotPresent
    name: init-container-1
    resources:
      limits:
        cpu: "1"
        memory: 1Gi
      requests:
        cpu: "1"
        memory: 1Gi
```

Since numa1 has  3 cores reserved for system components, i expect the pod could run successfully in numa2, but the pod got `TopologyAffinityError`, which is unexpected.  

Analisys:
1. `init container` run Admit function  before common containers, numa1 has enough cpu and is `Narrowest`, so it choose cpu numa1,  and set `reusableCPUs ` in [L242](https://github.com/kubernetes/kubernetes/blob/aae07c6f7bb715750a2d0cc77cf81c4e31a4904b/pkg/kubelet/cm/cpumanager/policy_static.go#L242)
2. `container-1` run Admit function, but requires to use the `reusableCPUs` in [L539](https://github.com/kubernetes/kubernetes/blob/aae07c6f7bb715750a2d0cc77cf81c4e31a4904b/pkg/kubelet/cm/cpumanager/policy_static.go#L539), so `container-1` can only goes to numa1, which do not have enough cpu to run , then `container-1` runs into `TopologyAffinityError`
3. After i set `init container`  request 14 Core CPUs and update the pod, pod success become running in numa2, which is expected.

To sum up, reuse cpus of init container could cause
1.  pod `TopologyAffinityError` because the numa node init container choose do not have enough resource for common container when kubelet run with `topologyManagerPolicy: single-numa-node`
2. pod  did not choose the `bestHint` when kubelet run with `topologyManagerPolicy: best-effort`

### Kubernetes version

v1.18.18

### Cloud provider

no

### OS version

Linux  5.4.119-19-0008.

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by xrmzju** [comment]
@klueska would you mind take a look at this issue please？

**Comment by klueska** [comment]
This is a known bug with a fairly simple workaround:
https://github.com/kubernetes/kubernetes/issues/94220#issuecomment-868489201

**Comment by xrmzju** [comment]
> This is a known bug with a fairly simple workaround: [#94220 (comment)](https://github.com/kubernetes/kubernetes/issues/94220#issuecomment-868489201)

sorry i miss that.  By the way,  would this issue be solved by set `topology-manager-scope`  to `pod`?
